tests/unit/jobserver/commands/test_users.py

```diff
@@ -90,3 +91,21 @@ def test_update_roles():
 
     user.refresh_from_db()
     assert user.roles == [ProjectDeveloper]
+
+
+def test_clear_all_roles(project_membership):
+    updator = UserFactory()
+    user = UserFactory(roles=[ProjectDeveloper])
+    OrgMembershipFactory(user=user, roles=[ProjectDeveloper])
+    project_membership(user=user, roles=[ProjectDeveloper])
+
+    assert len(user.roles) == 1
+    assert len(user.org_memberships.get().roles) == 1
+    assert len(user.project_memberships.get().roles) == 1
+
+    users.clear_all_roles(user=user, by=updator)
+
+    user.refresh_from_db()
+    assert len(user.roles) == 0
+    assert len(user.org_memberships.get().roles) == 0
+    assert len(user.project_memberships.get().roles) == 0
```

Comment:
I'm not completely sure about this, but is it worth asserting that there's an entry in the audit log here too?

Reply:
I was in two minds, because we'd be testing that there was an entry in the audit log twice. Once here, and once in `test_project_members.test_update_roles`. I agree, however, that _not_ testing doesn't feel quite right.

I guess it doesn't feel quite right because you'd have to know how `clear_all_roles` worked to know that it added an entry. In the future, we may rewrite `clear_all_roles` but forget to add an entry. `test_clear_all_roles` would pass, but `clear_all_roles` would be defective.

So yes, you're quite right. It's worth asserting that there's an entry in the audit log 🙂 Thank you for the nudge.